### PR TITLE
Adds parallelized version of Fratar

### PR DESCRIPTION
**Pull request by pedrocamargo**
Fratar (and by coincidence the calibration of synthetic gravity models) is too slow for large matrices. Multi-threading IPF and guaranteeing that everything runs on memory should solve the issue

### CODE REVIEW

File: aequilibrae/distribution/ipf.py [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
+
 import numpy as np
 import yaml
+
+from aequilibrae import global_logger
+from aequilibrae.context import get_active_project
 from aequilibrae.matrix import AequilibraeMatrix, AequilibraeData
 from aequilibrae.project.data.matrix_record import MatrixRecord
-from aequilibrae.context import get_active_project
+
```
Comment (pedrocamargo) [inline aequilibrae/distribution/ipf.py:14]: @jamiecook and @djfrancesco Does any of you have any idea of why do I have to wrap this in a try/except in order for it to build on Linux and MacOS?

Comment (djfrancesco) [reply aequilibrae/distribution/ipf.py:14]: It seems that the error originates from the `__init__.py` file in the aequilibrae subfolder. There is this import statement: `from aequilibrae import distribution` wether `compiled` is True or False. If we move it after `if compiled`, the try/except is not required anymore

File: aequilibrae/distribution/ipf_core.pyx [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+cpdef void _total_prods(double[:, :] flows,
+                        double[:] prod_tgt,
+                        double[:] prod_tot,
+                        int cpus)nogil:
+
+    cdef long long i, j
+    cdef long long I = flows.shape[0]
+    cdef long long J = flows.shape[1]
+
+    # Computes factors
```
Comment (jamiecook) [inline aequilibrae/distribution/ipf_core.pyx:142]:
```suggestion
    # Calculate the row totals (prods) from the flows
```

File: aequilibrae/distribution/setup_ipf.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    ext_modules = [
+        Extension(
+            "ipf_core",
+            ["ipf_core.pyx"],
+            extra_compile_args=["-fopenmp"],  # do we want -Ofast?
+            extra_link_args=["-fopenmp"],
+            define_macros=[("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION")],
+            include_dirs=[np.get_include()],
+        )
+    ]
```
Comment (jamiecook) [inline aequilibrae/distribution/setup_ipf.py:33]: open_
```suggestion
prefix = "/" if "WINDOWS" in platform.platform().upper() else "-f"
    
ext_modules = [
    Extension(
        "ipf_core",
        ["ipf_core.pyx"],
        extra_compile_args=[f'{prefix}openmp'],
        extra_link_args=[f'{prefix}openmp'],
        define_macros=[("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION")],
        include_dirs=[np.get_include()],
    )
]
```

**Review by jamiecook — APPROVED** [review]

**Pull request merged**